sphinx/tutorial/rose/furthertopics/failif-warnif.rst

```diff
@@ -0,0 +1,240 @@
+.. include:: ../../../hyperlinks.rst
+   :start-line: 1
+
+Fail-If, Warn-If
+================
+
+Basic validation can be achieved using metadata settings such as ``type`` and
+``range``. The ``fail-if`` and ``warn-if`` metadata settings are scriptable
+enabling more advanced validation, they evaluate logical expressions,
+flagging warnings if they return false.
+
+``fail-if`` and ``warn-if`` can be run on the command line using ``rose macro``
+or on-demand in the ``rose edit`` GUI.
+
+.. note::
+
+   Simple metadata settings such as ``range`` can be evaluated on-the-fly when
+   a value changes. As ``fail-if`` and ``warn-if`` can take longer to evaluate
+   they must be done on-demand in the ``rose edit`` GUI.
+
+
+Syntax
+------
+
+The syntax is Pythonic, and relies on `Jinja2`_ to actually evaluate
+relationships between values, after some initial pre-processing.
+
+You can reference setting values by using their ids - for example:
```

Comment:
~~ids~~ IDs is more recognisable/customary (would also need amending on line 34). Also I would flesh out this example a little bit with an explicit description e.g. 'for example the following references the settings for the metadata section "coffee", flagging an error if the value for the "cup_volume" setting is less than that for the "machine_output_volume" setting.'